src/aws.ts

```diff
@@ -504,6 +507,237 @@ export namespace $AWS {
       },
     });
   }
+
+  export const SDK: StepFunctionSDK = new Proxy<any>(
+    {},
+    {
+      get(_, serviceName: ServiceKeys) {
+        return new Proxy<any>(
+          {},
+          {
+            get: (_, methodName: string) => {
+              const defaultServicePrefix = serviceName.toLowerCase();
+              const defaultMethod =
+                methodName.charAt(0).toUpperCase() + methodName.slice(1);
+              const defaultIamActions = [
+                `${defaultServicePrefix}:${defaultMethod}`,
+              ];
+
+              return makeIntegration<
+                `$AWS.SDK.${ServiceKeys}`,
+                (input: any) => Promise<any>
+              >({
+                kind: `$AWS.SDK.${serviceName}`,
+                native: {
+                  bind(context, args) {
+                    const iamOptions = args[1] as unknown as IamOptions;
+
+                    if (!isObjectLiteralExpr(iamOptions)) {
+                      throw new SynthError(
+                        ErrorCodes.Expected_an_object_literal,
+                        `Second argument ('iamOptions') into a SDK call must be an object`
+                      );
+                    }
+
+                    context.resource.addToRolePolicy(
+                      new aws_iam.PolicyStatement({
+                        effect: aws_iam.Effect.ALLOW,
+                        actions: iamOptions.iamActions ?? defaultIamActions,
+                        resources: iamOptions.iamResources,
+                        conditions: iamOptions.iamConditions,
+                      })
+                    );
+                  },
+                  preWarm(preWarmContext) {
+                    preWarmContext.getOrInit({
+                      key: `$AWS.SDK.${serviceName}`,
+                      init: (key, props) =>
+                        new AWS[serviceName](
+                          props?.clientConfigRetriever?.(key)
+                        ),
+                    });
+                  },
+                  call(args, preWarmContext) {
+                    const client: any = preWarmContext.getOrInit({
+                      key: `$AWS.SDK.${serviceName}`,
+                      init: (key, props) =>
+                        new AWS[serviceName](
+                          props?.clientConfigRetriever?.(key)
+                        ),
+                    });
+
+                    return client[methodName](args[0]).promise();
+                  },
+                },
+                asl: (call, context) => {
+                  const iamOptions = call.args[1] as unknown as IamOptions;
```

Comment:
This is actually the Argument, not an `IamOptions`, any value in typing this as `IamOptions` here?

Reply:
I wasn't sure how to get to the real options, thanks for the pointer to `evalToConstant` though 🙏